### ISSUE TITLE
Bug: length_scale_prior is silently ignored  kernel always uses the default (1e-5, 1e5) length-scale bounds

### ISSUE THREAD

**Issue by jonaselgammal** [jonaselgammal/GPry#3]
### Summary

The `length_scale_prior` passed to `GaussianProcessRegressor` never reaches the kernel. It is
handed to `length_corr_kernel` under the **`prior_bounds`** keyword instead of
`length_scale_bounds`, so `length_scale_bounds` silently falls back to its sklearn default
`(1e-5, 1e5)`.

Every GP fit on `main` therefore optimises the length scales over **10 decades** instead of the
intended **4** (`[1e-3, 1e1]`), and `prior_bounds` simultaneously receives an object it was never
meant to hold.

### Reproducer

```python
import numpy as np
from gpry.gpr import GaussianProcessRegressor

lsp = np.array([[1e-3, 1e1]] * 4)          # the documented default prior
g = GaussianProcessRegressor(kernel="RBF", length_scale_prior=lsp)
print(np.exp(g.kernel.bounds)[1:])         # -> [1e-05, 1e+05] per dimension, not [1e-3, 1e1]
print(g.kernel.k2.length_scale_bounds)     # -> (1e-05, 100000.0)  i.e. the untouched default
```

### Root cause

`gpry/gpr.py` (currently ~L201):

```python
) * length_corr_kernel(
    length_scale_init,
    prior_bounds=length_scale_prior,   # <-- wrong keyword
    **kernel_args,
)
```

`RBF.__init__(self, length_scale=1.0, length_scale_bounds=(1e-5, 1e5), prior_bounds=None)` —
`length_scale_bounds` is never passed, so it keeps its default. `prior_bounds` is only consumed
when `length_scale_bounds == "dynamic"` (to compute `max_length`), and `"dynamic"` is not
requested anywhere in the codebase, so the value passed there is inert.

Introduced in **39dc81c** *"split surrogate and gpr: first working (imperfectly) implementation"*
(2025-07-03), the surrogate/GPR split:

```diff
-                ) * length_corr_kernel(
-                    [length_scale_init] * self.d,
-                    length_scale_prior,           # positional -> length_scale_bounds  (correct)
-                    prior_bounds=self.bounds_,    # parameter-space bounds             (correct)
+            ) * length_corr_kernel(
+                length_scale_init,
+                prior_bounds=length_scale_prior,  # wrong slot
```

### Suggested fix

Keep the `length_scale_prior` vocabulary throughout (clearer, and it matches the public API), and
restore the parameter-space bounds so `"dynamic"` works again:

* `length_scale_prior` — the prior on the length-scale *hyperparameter*;
* `prior_bounds` — the *parameter-space* box, in the kernel's (transformed) input space, used to
  derive `max_length` when `length_scale_prior == "dynamic"`.

1. **`gpry/kernels.py`** — rename the constructor argument and attribute
   `length_scale_bounds` → `length_scale_prior` in all four affected kernels, updating the
   `hyperparameter_length_scale` overrides accordingly, and set the default to `(1e-3, 1e1)`.
   This is safe: GPry never calls `super().__init__()` in these kernels and already overrides
   `hyperparameter_length_scale`; sklearn touches `length_scale_bounds` only in those two places,
   and `get_params`/`clone` introspect GPry's own signature. Keep `prior_bounds` as-is.

2. **Guard the four `"dynamic"` comparisons** so an array-valued prior cannot explode:

   ```python
   if isinstance(length_scale_prior, str) and length_scale_prior == "dynamic":
   ```

   This is what forced the original workaround: `length_scale_prior` is now per-dimension, shape
   `(d, 2)`, and `array == "dynamic"` raises.

3. **`gpry/gpr.py`** — give the GPR back the parameter-space bounds as a new optional argument
   (they were lost when `self.bounds_` moved to `SurrogateModel`), and pass both quantities under
   their correct names:

   ```python
   def __init__(self, kernel="RBF", ..., n_restarts_optimizer=0,
                prior_bounds=None,          # NEW: transformed parameter-space bounds
                random_state=None):
   ...
   ) * length_corr_kernel(
       length_scale_init,
       length_scale_prior=length_scale_prior,
       prior_bounds=prior_bounds,
       **kernel_args,
   )
   ```

4. **`gpry/surrogate.py`** — supply it. The required value is already computed in
   `SurrogateModel.__init__` (currently ~L166, for the infinities classifier):

   ```python
   bounds_ = self.preprocessing_X.transform_bounds(self._bounds)
   ...
   kwargs_regressor["prior_bounds"] = bounds_
   ```

   Hoisting that line above the regressor construction is enough; `NormalizeBounds` sets
   `fitted = True` at init, so no ordering problem arises. This reproduces the pre-refactor
   semantics exactly (`prior_bounds = preprocessing_X.transform_bounds(bounds)`), which is the
   transformed space the kernel — and hence `max_length` — lives in.

   The kernels already raise a clear `TypeError` when `"dynamic"` is requested without
   `prior_bounds`, so misuse stays loud.
